### ISSUE TITLE
Use TrackedDownloadState instead of TrackedDownloadStatus in queue metric

### ISSUE THREAD

**Issue by onedr0p** [onedr0p/radarr-exporter#8]
Hopefully I'll be able to grab the TrackedDownloadState from the API once this PR gets merged Radarr/Radarr#4257 and group by the below fields.

```
public enum TrackedDownloadState
{
        Downloading,
        ImportPending,
        Importing,
        Imported,
        FailedPending,
        Failed,
        Ignored
}
```

Related issue #3

**Comment by rubasace** [comment]
Regarding the previous issue, I think OK/Warning is good enough as long as errors come as Warning. It makes it easier to add an alert rule checking if things are !=OK. I cannot think of any warning that isn't important and not sure how from the metrics point of view it might be relevant.

This might not be the case for you/others, but then I'd probably add a numeric value for the new enum, so alerting can be done in a single rule bu comparing the numeric value

**Comment by onedr0p** [comment]
If I am thinking correct Errors are in fact listed as Warnings, so you logic of `!=Ok` is valid. Please let me know if you find this is not the case.

Once those changes are merged into Radarr I will revisit this and hopefully be able to break out the metrics even more.

**Comment by rubasace** [comment]
So far it seems to be the case, so I'm happy as it is :)

**Comment by onedr0p** [comment]
I will try to implement this into Sonarr the next chance I get, because they already have that code :)

**Comment by onedr0p** [comment]
**Edit** Moved comment to it's own issue https://github.com/onedr0p/radarr-exporter/issues/10

**Comment by onedr0p** [comment]
I am giving this a go in my sonarr-exporter and sadly the JSON does not contain the type of `TrackedDownloadState`

Here's a snippet of the JSON request

```json
[
  {
    "series": {
        ...
    }
    ...
    "status": "Completed",
    "trackedDownloadStatus": "Warning",
    "statusMessages": [
      {
        "title": "b793b233ae944e1e8610fe89f3ad2ac3.mkv",
        "messages": [
          "This show has individual episode mappings on TheXEM but the mapping for this episode has not been confirmed yet by their administrators. TheXEM needs manual input."
        ]
      }
    ],
    ...
  }
]
```

This issue will be put on hold until the API gives us this information.

**Comment by onedr0p** [comment]
I have opened issue with Sonar team ^

**Comment by onedr0p** [comment]
This code is written and is in the 2.0.0 release, however we're still waiting on Radarr to get it merged
